src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.addAlive(this);
+  }
+
+  static addAlive(animal) {
+    Animal.alive.push(animal);
+  }
+
+  static removeDead() {
+    return Animal.alive.filter(animal => animal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = hidden || false;
+  }
```

Comment:
The same situation as at previous comment.

```suggestion
  constructor(name, health, hidden = false) {
    super(name, health);
    this.hidden = hidden;
  }
```